issue_manager_script.py

```diff
@@ -127,114 +92,125 @@
         project = p  # Assign the found project
         break        # Exit the loop once the project is found
 
-# If the project was not found, exit the script
 if not project:
     print(f"Project number {project_number} not found.")
     exit(1)
 
-# Edge Case 16: Prevent overlapping runs
-# Create a lock file to ensure only one instance runs at a time
-lock_file = '/tmp/why_drs_issue_manager.lock'
+# ----------------------- Prevent Overlapping Runs ----------------------------
 
+# Edge Case 16: Concurrency and Race Conditions
+# Implement a lock file mechanism to prevent overlapping runs
 if os.path.exists(lock_file):
-    print("Another instance of the script is already running. Exiting to prevent overlap.")
-    exit(1)
+    # Check if the lock file is older than 1 hour
+    if time.time() - os.path.getmtime(lock_file) > 3600:
+        os.remove(lock_file)  # Remove stale lock file
+    else:
+        print("Another instance of the script is already running. Exiting to prevent overlap.")
```

Comment:
I can't think of why this might happen, except for that I am not completely familiar with exactly where these scripts run. Are there single execution points for all of the workflow scripts (in Azure) or somewhere we can reference? 🔩